### PR TITLE
[stable/datadog] allow host networking for metrics

### PR DESCRIPTION
**Pull request by winjer**
Signed-off-by: Doug Winter <doug.winter@isotoma.com>

#### What this PR does / why we need it:

In CNI situations where the masters are unable to use CNI networking and you wish to make external metrics available, the metrics server needs to support host networking.

This PR includes the appropriate changes and values.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/datadog/templates/agent-service-metrics.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ metadata:
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
 spec:
+  type: {{ .Values.clusterAgent.service.type }}
```
Comment (clamoriniere) [inline stable/datadog/templates/agent-service-metrics.yaml]: Default value of type is `clusterIP`, so I think we can overwrite the default value only `serviceType` present in the `values.yaml` file.

```suggestion
  {{- if .Values.clusterAgent.serviceType }}
  type: {{ .Values.clusterAgent.serviceType }}
  {{- end }}
```

File: stable/datadog/values.yaml [hunk 2]
```diff
@@ -360,6 +360,23 @@ clusterAgent:
   #
   # readinessProbe:
 
+  ## @param service - object - required
```
Comment (clamoriniere) [inline stable/datadog/values.yaml]: I would prefer to have this parameter optional 
```suggestion
  ## @param service - object - optional
  ##
  # serviceType: ClusterIP
```
And I think that `annotations` is not used.

**Review by clamoriniere — COMMENTED** [review]
Hi @winjer 
Thanks you for opening this pull request.
let me know if my comments make sense to you.

File: stable/datadog/values.yaml [hunk 3]
```diff
@@ -360,6 +360,25 @@ clusterAgent:
...
   # readinessProbe:
 
+  ## @param service - object - optional
+  ##
+  # serviceType: ClusterIP
+  ##
+  #
+  service:
+    type: ClusterIP
+    annotations: {}
```
Comment (clamoriniere) [inline stable/datadog/values.yaml]: why did you add `annotations: {}`?
I didn't see where it is used.

Comment (winjer) [reply stable/datadog/values.yaml]: you are correct, that's a mistake. I have removed it.

**Comment by clamoriniere** [comment]
could you please bump the Chart version and fix the conflict? 
Then I will be able to merge you PR.

File: stable/datadog/templates/agent-services.yaml [hunk 4]
```diff
@@ -30,6 +30,9 @@ metadata:
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
 spec:
+  {{- if .Values.clusterAgent.serviceType }}
```
Comment (hkaj) [inline stable/datadog/templates/agent-services.yaml]:
```suggestion
  {{- if .Values.clusterAgent.service.type }}
```

File: stable/datadog/templates/agent-services.yaml [hunk 5]
```diff
@@ -30,6 +30,9 @@ metadata:
...
+  type: {{ .Values.clusterAgent.serviceType }}
```
Comment (hkaj) [inline stable/datadog/templates/agent-services.yaml]:
```suggestion
  type: {{ .Values.clusterAgent.service.type }}
```

File: stable/datadog/values.yaml [hunk 6]
```diff
@@ -371,6 +371,24 @@ clusterAgent:
   #
   # readinessProbe:
 
+  ## @param service - object - optional
+  ##
+  # serviceType: ClusterIP
```
Comment (hkaj) [inline stable/datadog/values.yaml]: This should be directly on top of `type: ClusterIP`, and prefixed with
`## @param type - string - optional`

**Review by hkaj — CHANGES_REQUESTED** [review]
sorry for the delay @winjer, could I left some comments, could you address them, and bump the chart version before we merge?

**Comment by plumdog** [comment]
@clamoriniere @hkaj Hello - I've picked this up from @winjer.

I've updated based on comments, though I've also moved the new service config to be under `clusterAgent.metricsProvider` because there are two cluster agent services, and this setting only alters the metrics one, so I think this is clearer.

I think the docstrings in `values.yaml` are now correctly formatted and positioned.

**Comment by plumdog** [comment]
@clamoriniere @hkaj I've rebased to resolve the conflict and to remove the stale label.

I believe I'd already made all of the requested changes.

**Comment by clamoriniere** [comment]
@plumdog thanks for updating your PR.
I will do a final review, before merging it.

File: stable/datadog/values.yaml [hunk 7]
```diff
@@ -356,6 +356,15 @@ clusterAgent:
   metricsProvider:
     enabled: false
 
+    ## @param service - object - optional
```
Comment (clamoriniere) [inline stable/datadog/values.yaml]: This line can be removed

**Review by clamoriniere — COMMENTED** [review]
just a small change linked to this previous comment: https://github.com/helm/charts/pull/15206#discussion_r317094379

File: stable/datadog/values.yaml (hunk 7 above)
Comment (plumdog) [reply stable/datadog/values.yaml]: Fixed.

Comment (clamoriniere) [reply stable/datadog/values.yaml]: thanks 👍

**Comment by clamoriniere** [comment]
@plumdog thanks for your contribution 👍
